Riot/Managers/PushRulesUpdater/PushRulesUpdater.swift

```diff
@@ -0,0 +1,81 @@
+// 
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Combine
+
+final class PushRulesUpdater {
+    private var cancellables: Set<AnyCancellable> = .init()
+    private var rules: [NotificationPushRuleType] = []
+    private let notificationSettingsService: NotificationSettingsServiceType
+    private let didCompleteUpdateSubject: PassthroughSubject<Void, Never> = .init()
+    
+    var didCompleteUpdate: AnyPublisher<Void, Never> {
+        didCompleteUpdateSubject.eraseToAnyPublisher()
+    }
```

Comment:
Is this unused? 🤔

Reply:
It had a purpose just for unit tests.
Deleted as a part of this: https://github.com/vector-im/element-ios/pull/7335/commits/3972da892e9dd6362dd70990e408bea7afaca419